lib/validator/validator.go

```diff
@@ -520,6 +554,66 @@ func validateBGPPeerMeta(v *validator.Validate, structLevel *validator.StructLev
 	}
 }
 
+func validateBackendEndpointPort(v *validator.Validate, structLevel *validator.StructLevel) {
```

Comment:
All this boilerplate (also the code above which checks for duplicate named ports for both `WorkloadEndpointSpec` and `HostEndpointSpec`) ... I'll let more experienced golang devs comment on this, but it does look verbose.